app/controllers/posts_controller.rb

```diff
@@ -7,4 +7,26 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    post_params = params.require(:post).permit(:title, :text)
+    @post = Post.new(title: post_params[:title], text: post_params[:text], comments_counter: 4, likes_counter: 9,
+                     author_id: current_user.id)
+
```

Comment:
- [ ] You have implemented strong params, but it had to be defined as a private method in posts_controller.
```
   private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

then just pass it in arguments of Post.new: 
`Posts.new(post_params)`

- [ ] I can see that you are using params post_params and listing all arguments. Rails is smart, if you just put `Posts.new(post_params)` it will know what to do with params and automatically assign these values, you don't have to be explicit. 

Another issue is how would you then assign a user id to this post right? Its easy due to associations in model, post belongs to user, so you simply should 

`@post = current_user.posts.new(post_params)`

full code to avoid your confusion: 
```ruby
# controllers/posts_controller.rb

  class PostsController < ApplicationController

    ...
def create
    @post = current_user.posts.new(post_params)
    @post.likes = 0
    @post.comments = 0
    
    respond_to do |format|
      format.html do
        if @post.save
          flash[:success] = 'Post saved successfully'
          redirect_to user_posts_path(current_user.id)
        else
          puts @post.errors.full_messages
          flash[:error] = 'Post could not be saved'
          render :new, locals: { post: @post }
        end
      end
    end
  end

    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
	  ...
  end
```